core/javatests/com/google/instrumentation/stats/ViewDescriptorTest.java

```diff
@@ -32,33 +34,70 @@
 @RunWith(JUnit4.class)
 public final class ViewDescriptorTest {
   @Test
-  public void testViewDescriptor() {
+  public void testDistributionViewDescriptor() {
+    String name = "test-view-name";
+    String description = "test-view-name description";
+    MeasurementDescriptor mDescriptor = MeasurementDescriptor.create(
+        "measurement",
+        "measurement description",
+        MeasurementUnit.create(1, Arrays.asList(new BasicUnit[] { BasicUnit.SCALAR })));
+    DistributionAggregationDescriptor dAggrDescriptor = DistributionAggregationDescriptor.create();
+    List<TagKey> keys = Arrays.asList(new TagKey[] { new TagKey("foo"), new TagKey("bar") });
+    final DistributionViewDescriptor viewDescriptor =
+        DistributionViewDescriptor.create(name, description, mDescriptor, dAggrDescriptor, keys);
+
+    assertThat(viewDescriptor.getName()).isEqualTo(name);
+    assertThat(viewDescriptor.getDescription()).isEqualTo(description);
+    assertThat(viewDescriptor.getMeasurementDescriptor().getName()).isEqualTo(mDescriptor.getName());
+    assertThat(viewDescriptor.getDistributionAggregationDescriptor()).isEqualTo(dAggrDescriptor);
+    assertThat(viewDescriptor.getTagKeys().size()).isEqualTo(2);
+    assertThat(viewDescriptor.getTagKeys().get(0).toString()).isEqualTo("foo");
+    assertThat(viewDescriptor.getTagKeys().get(1).toString()).isEqualTo("bar");
+    assertThat(viewDescriptor.match(
+        new Function<DistributionViewDescriptor, Boolean> () {
+          @Override public Boolean apply(DistributionViewDescriptor dViewDescriptor) {
+            return dViewDescriptor == viewDescriptor;
+          }
+        },
+        new Function<IntervalViewDescriptor, Boolean> () {
+          @Override public Boolean apply(IntervalViewDescriptor iViewDescriptor) {
+            return false;
+          }
+        })).isTrue();
+  }
+
+  @Test
+  public void testIntervalViewDescriptor() {
```

Comment:
Could this test share the MeasurementDescriptor and tags with the previous test?